qiskit/transpiler/passmanager.py

```diff
@@ -541,3 +424,29 @@ def draw(self, filename=None, style=None, raw=False):
         from qiskit.visualization import staged_pass_manager_drawer
 
         return staged_pass_manager_drawer(self, filename=filename, style=style, raw=raw)
+
+
+# A temporary error handling with slight overhead at class loading.
+# This method wraps all class methods to replace PassManagerError with TranspilerError.
+# The pass flow controller mechanics raises PassManagerError, as it has been moved to base class.
+# PassManagerError is not caught by TranspilerError due to the hierarchy.
+
+
+def _replace_error(meth):
+    @wraps(meth)
+    def wrapper(*meth_args, **meth_kwargs):
+        try:
+            return meth(*meth_args, **meth_kwargs)
+        except PassManagerError as ex:
+            raise TranspilerError(ex.message) from ex
+
+    return wrapper
+
+
+for _name, _method in inspect.getmembers(PassManager, predicate=inspect.isfunction):
```

Comment:
I didn't try detailed profiling and ASV benchmark, but likely it just causes a slight overhead at module loading (percentage is too small to be detected). Since this only wraps public methods (such as `.run`), performance regression during run time is not detected. 

![image](https://github.com/Qiskit/qiskit-terra/assets/39517270/abcfe061-a463-4609-af3f-78af1731818e)

If this causes a visible regression in the ASV, I'll open followup PR.